datastream/tools/split_dataframes.py

```diff
@@ -29,43 +29,50 @@ def split_dataframes(
         ]))
     
     if filepath is not None and filepath.exists():
-        split = json.read(filepath)
+        split = json.loads(filepath.read_text())
 
         if set(proportions.keys()) != set(split.keys()):
             raise ValueError(
-                'Expected split names in split file to be the same as the keys'
-                'of proportions'
+                'Expected split names in split file to be the same as the'
+                'keys of proportions'
             )
     else:
         split = {
             split_name: list()
             for split_name in proportions.keys()
         }
 
-    split_proportions = tuple(proportions.items())
+    if len(dataframe[key_column].unique()) != len(dataframe):
+        raise KeyError(f'key_column {key_column} contains duplicate values')
```

Comment:
https://docs.python.org/3/library/exceptions.html#KeyError
is pretty specific

I think this is more of a ValueError

Reply:
Maybe I don't care that much but I think this is quite literally a key error? Would you call it a key error if the dataframe was a dictionary instead? I'm open to changing this

Reply:
`key is not found in the set of existing keys` is pretty specific (any mapping type), but if the mapping is not on the form expected its something else